### PR TITLE
Feat/adaptive

### PR DESCRIPTION
**Pull request by dandgerson**

### CODE REVIEW

File: app/_variables.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+$grid-breakpoints: (
```
Comment (komarov-sergey) [inline app/_variables.scss:1]: nice!

File: app/components/button/button.tsx [hunk 2]
```diff
@@ -9,14 +9,16 @@ const Button: FC<{
   outline?: boolean
   main?: boolean
   style?: Record<string, string | number>
-}> = ({ children, outline, main, style = {} }) => {
+  onClick?: MouseEventHandler<HTMLDivElement> | undefined
+}> = ({ children, outline, main, style = {}, onClick }) => {
   return (
     <div
       className={clsx(s.button, {
         [s['button-outline']]: outline,
```
Comment (komarov-sergey) [inline app/components/button/button.tsx:17]: difficult naming for .scss

File: app/header.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import Link from 'next/link'
+import { Navbar } from './components/navbar/navbar'
+import { FC, useEffect, useRef, useState } from 'react'
+import { usePathname } from 'next/navigation'
+
+export const Header: FC<{ s: Record<string, string> }> = ({ s }) => {
+  const [offset, setOffset] = useState(0)
+  const pathname = usePathname()
+
+  useEffect(() => {
```
Comment (komarov-sergey) [inline app/header.tsx:14]: maybe change order and do useState before useEffects?

File: app/layout.module.scss [hunk 4]
```diff
@@ -1,36 +1,34 @@
...
-    align-items: center;
-    gap: 2rem;
-  }
-
   &_buttons {
     display: flex;
     gap: 1rem;
   }
+
+  @media (max-width: map.get($grid-breakpoints, 'lg')) {
```
Comment (komarov-sergey) [inline app/layout.module.scss:20]: media query not on top level it is good practise?

File: app/layout.tsx [hunk 5]
```diff
@@ -1,119 +1,89 @@
...
+                {
+                  path: '/try-free',
+                  label: 'Получить годовую лицензию',
+                },
+              ]}
+            />
+          </nav>
+
+          <div className={s.content_item}>
+            © 2021-2023.ООО «ЧТО ТО», официальный сайт in2sql. Передавая нам
```
Comment (komarov-sergey) [inline app/layout.tsx]: it's better to get final year dynamically like new Date().getFullYear()

**Review by komarov-sergey — APPROVED** [review]

File: app/components/button/button.tsx (hunk 2 above)
Comment (dandgerson) [reply app/components/button/button.tsx:17]: camon bro, it's just name of a class, let them kill itself as not the best practice which no one use it =))

File: app/header.tsx (hunk 3 above)
Comment (dandgerson) [reply app/header.tsx:14]: in this one we grouping side effects by it's owned state variables, so it's more clear for identify what own's each other =)

File: app/layout.module.scss (hunk 4 above)
Comment (dandgerson) [reply app/layout.module.scss:20]: it depends, bro =)
in this case i use it for more specific rule for header =)

That's what the AI answered))


>Best Practice:
Consider adopting a hybrid approach that strikes a balance between the two. Keep simple and generic media queries at the top level, and for more complex or specific cases, consider nesting them within relevant CSS rules.

>Additionally, using a CSS preprocessor like Sass or LESS can provide more organization and maintainability through the use of mixins or functions for media queries.

>Ultimately, consistency and readability should be prioritized. Choose an approach that makes your CSS codebase maintainable and easy for you and your team to work with.Best Practice:
Consider adopting a hybrid approach that strikes a balance between the two. Keep simple and generic media queries at the top level, and for more complex or specific cases, consider nesting them within relevant CSS rules.

>Additionally, using a CSS preprocessor like Sass or LESS can provide more organization and maintainability through the use of mixins or functions for media queries.

>Ultimately, consistency and readability should be prioritized. Choose an approach that makes your CSS codebase maintainable and easy for you and your team to work with.

**Pull request merged**